src/plugins/workspace/public/components/workspace_menu/workspace_menu.tsx

```diff
@@ -0,0 +1,191 @@
+  EuiPopover,
+  EuiText,
+} from '@elastic/eui';
+import type { EuiContextMenuPanelItemDescriptor } from '@elastic/eui';
+
+import {
+  WORKSPACE_CREATE_APP_ID,
+  WORKSPACE_LIST_APP_ID,
+  WORKSPACE_OVERVIEW_APP_ID,
+} from '../../../common/constants';
+import { cleanWorkspaceId, formatUrlWithWorkspaceId } from '../../../../../core/public/utils';
+import { CoreStart, WorkspaceObject } from '../../../../../core/public';
+
+interface Props {
+  coreStart: CoreStart;
+}
+
+/**
+ * Return maximum five workspaces, the current selected workspace
+ * will be on the top of the list.
+ */
+function getFilteredWorkspaceList(
+  workspaceList: WorkspaceObject[],
+  currentWorkspace: WorkspaceObject | null
+): WorkspaceObject[] {
+  return [
+    ...(currentWorkspace ? [currentWorkspace] : []),
+    ...workspaceList.filter((workspace) => workspace.id !== currentWorkspace?.id),
+  ].slice(0, 5);
+}
+
+export const WorkspaceMenu = ({ coreStart }: Props) => {
+  const [isPopoverOpen, setPopover] = useState(false);
+  const currentWorkspace = useObservable(coreStart.workspaces.currentWorkspace$, null);
+  const workspaceList = useObservable(coreStart.workspaces.workspaceList$, []);
+
+  const defaultHeaderName = i18n.translate(
+    'core.ui.primaryNav.workspacePickerMenu.defaultHeaderName',
+    {
+      defaultMessage: 'Select a workspace',
+    }
+  );
+  const filteredWorkspaceList = getFilteredWorkspaceList(workspaceList, currentWorkspace);
+  const currentWorkspaceName = currentWorkspace?.name ?? defaultHeaderName;
+
+  const openPopover = () => {
+    setPopover(!isPopoverOpen);
+  };
+
+  const closePopover = () => {
+    setPopover(false);
+  };
+
+  const workspaceToItem = (workspace: WorkspaceObject) => {
+    const workspaceURL = formatUrlWithWorkspaceId(
+      coreStart.application.getUrlForApp(WORKSPACE_OVERVIEW_APP_ID, {
+        absolute: false,
+      }),
+      workspace.id,
+      coreStart.http.basePath
+    );
+    const name =
+      currentWorkspace?.name === workspace.name ? (
+        <EuiText>
+          <strong>{workspace.name}</strong>
+        </EuiText>
+      ) : (
+        workspace.name
+      );
+    return {
+      name,
+      key: workspace.id,
+      icon: <EuiIcon type="stopFilled" color={workspace.color ?? 'primary'} />,
+      onClick: () => {
+        window.location.assign(workspaceURL);
+      },
+    };
+  };
+
+  const getWorkspaceListItems = () => {
+    const workspaceListItems: EuiContextMenuPanelItemDescriptor[] = filteredWorkspaceList.map(
+      workspaceToItem
+    );
+    workspaceListItems.push({
+      icon: <EuiIcon type="plus" />,
+      name: i18n.translate('core.ui.primaryNav.workspaceContextMenu.createWorkspace', {
+        defaultMessage: 'Create workspace',
+      }),
+      key: WORKSPACE_CREATE_APP_ID,
+      onClick: () => {
+        window.location.assign(
+          cleanWorkspaceId(
+            coreStart.application.getUrlForApp(WORKSPACE_CREATE_APP_ID, {
+              absolute: false,
```

Comment:
what does absolute:false do?

Reply:
If set to true, it will return an absolute url instead of a relative one

Reply:
Got it, thanks